lib/algolia.ex

```diff
@@ -163,7 +169,8 @@ defmodule Algolia do
       path_encode_fun: &URI.encode/1,
       connect_timeout: 3_000 * (curr_retry + 1),
       recv_timeout: 30_000 * (curr_retry + 1),
-      ssl_options: [{:versions, [:"tlsv1.2"]}]
+      ssl_options: [{:versions, [:"tlsv1.2"]}],
+      pool: pool_name()
```

Comment:
Not sure if making this configurable would bring much value. I'd rather keep it hardcoded like the other hackney opts we already have in place. IMO the chances of having a conflicting pool name across the app are too small to make it worth a config now.

Reply:
In addition to that, adding yet another `Application.get_env`-based config will make our life harder in the future when we eventually move to non-global configs, as [recommended](https://hexdocs.pm/elixir/library-guidelines.html#avoid-application-configuration).